CardScan/Classes/Extensions/Image+utils.swift

```diff
@@ -35,51 +35,44 @@ extension CGImage {
-        // The size of the cropped region is needed to map regionOfInterestCenter to the image center
-        let imageAspectRatio = CGFloat(self.width) / CGFloat(self.height)
-        let screenAspectRatio = screenWidth / screenHeight
-        
-        var pointsToPixels: CGFloat
-        // convert from points to pixels and account for the cropped region
-        if imageAspectRatio > screenAspectRatio {
-            // left and right of the image cropped
-            //      tested on: iPhone XS Max
-            let croppedOffset = (CGFloat(self.width) - CGFloat(self.height) * screenAspectRatio) / 2.0
-            pointsToPixels = CGFloat(self.height) / screenHeight
+    func toFullScreenAndRoi(previewViewFrame: CGRect, regionOfInterestLabelFrame: CGRect) -> (CGImage, CGRect)? {
+        let imageCenterX = CGFloat(self.width) / 2.0
+        let imageCenterY = CGFloat(self.height) / 2.0
             
-            cx = regionOfInterestCenterX * pointsToPixels + croppedOffset
-            cy = regionOfInterestCenterY * pointsToPixels
-        } else {
-            // top and bottom of the image cropped
-            //      tested on: iPad Mini 2
-            let croppedOffset = (CGFloat(self.height) - CGFloat(self.width) / screenAspectRatio) / 2.0
-            pointsToPixels = CGFloat(self.width) / screenWidth
+        let imageAspectRatio = CGFloat(self.height) / CGFloat(self.width)
+        let previewViewAspectRatio = previewViewFrame.height / previewViewFrame.width
+        let cropRatio = CGFloat(16.0) / CGFloat(9.0)
             
-            cx = regionOfInterestCenterX * pointsToPixels
-            cy = regionOfInterestCenterY * pointsToPixels + croppedOffset
-        }
-        
-        let roiWidthInPixels = regionOfInterestLabelFrame.size.width * pointsToPixels
-        let roiHeightInPixels = regionOfInterestLabelFrame.size.height * pointsToPixels
-        return CGRect(x: cx - roiWidthInPixels * 0.5,
-                      y: cy - roiHeightInPixels * 0.5,
-                      width: roiWidthInPixels,
-                      height: roiHeightInPixels)
+        // Get ratio to convert points to pixels
+        let pointsToPixel: CGFloat = imageAspectRatio > previewViewAspectRatio ? CGFloat(self.width) / previewViewFrame.width : CGFloat(self.height) / previewViewFrame.height
+          
+        let fullScreenImage: CGImage? = {
+            // if image is already 16:9, no need to crop to match crop ratio
+            if cropRatio == imageAspectRatio {
+                return self
+            } else {
+                // imageAspectRatio not being 16:9 implies image being in landscape
+                // get width to first not cut out any card information
+                let cropWidth = previewViewFrame.width * pointsToPixel
+                // then get height to match 16:9 ratio
+                let cropHeight = cropWidth * (16.0 / 9.0)
```

Comment:
There is a corner case here where the cropHeight might be too big for the image that we have that we can handle. If you're confident that we won't see it in practice yet, I suggest adding a `// todo(jaime): handle case where height is too big for the image` comment to track it and tackle it with the next pr.